### PR TITLE
feat(techdocs): use `edit_uri_template` instead of `edit_uri`

### PR DESCRIPTION
**Pull request by pjungermann**
Use [`edit_uri_template`](https://www.mkdocs.org/user-guide/configuration/#edit_uri_template) instead of [`edit_uri`](https://www.mkdocs.org/user-guide/configuration/#edit_uri) and remove integration type-based limit.

This features was introduced as part of
[mkdocs v1.4.0](https://github.com/mkdocs/mkdocs/releases/tag/1.4.0).

Effectively, this will add at least `repo_url` to all `url`-type locations of all integrations.

Additionally, `edit_uri_template` will only be added if the integration supports an edit URL
(== `resolveEditUrl` does not return URL unchanged).

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by pjungermann** [comment]
Could anyone reopen this?

**Review by Rugvip — APPROVED** [review]
Sorry, this one must've slipped off the radar.

Looks good to me! 👍 Will leave final say to @backstage/techdocs-core

**Comment by Rugvip** [comment]
Please mark this as ready for review if it is btw. Otherwise reviewers won't get pinged automatically and it will have to be discovered in other ways, which just happened 😅

**Comment by pjungermann** [comment]
@Rugvip I've kept it a draft for now as it requires an upgrade at the mkdocs plugins - and so far, I've struggled a bit with that 😅

**Comment by pjungermann** [comment]
not stale

**Comment by pjungermann** [comment]
not stale

**Comment by pjungermann** [comment]
no stale

**Comment by pjungermann** [comment]
no stale

**Comment by Rugvip** [comment]
Marked as will-fix, let is know when it's time to have a look 😁

File: plugins/techdocs-node/src/stages/generate/mkdocsPatchers.ts [hunk 1]
```diff
@@ -108,7 +109,12 @@ export const patchMkdocsYmlPreBuild = async (
   scmIntegrations: ScmIntegrationRegistry,
 ) => {
   await patchMkdocsFile(mkdocsYmlPath, logger, mkdocsYml => {
-    if (!('repo_url' in mkdocsYml) && !('edit_uri' in mkdocsYml)) {
+    console.log(mkdocsYmlPath);
```
Comment (johnphilip283) [inline plugins/techdocs-node/src/stages/generate/mkdocsPatchers.ts:112]:
```suggestion

```

**Comment by pjungermann** [comment]
Update:
mkdocs-techdocs-core already requires a recent enough mkdocs version to be able to use edit_uri_template instead of edit_uri.

However, mkdocs-monorepo-plugin still requires some changes (and does not yet require a recent enough version, yet).

**Comment by squid-ney** [comment]
Hi @pjungermann , some work was recently merged in PR https://github.com/backstage/backstage/pull/25549 to fix an issue with the `patchMkdocsYmlPreBuild` that caused merge conflicts here. 

Question about the solution proposed, if the `mkdocs.yml` doesn't include an `edit_uri` or `edit_uri_template`, what is the benefit of patching `edit_uri_template` over `edit_uri` in the `patchMkdocsYmlPreBuild`?

**Pull request closed without merging**